### PR TITLE
migrate augmentation to albumentations

### PR DESCRIPTION
**Pull request by amrukwa**

### CODE REVIEW

File: semantic_segmentation/scripts/augment.py [hunk 1]
```diff
@@ -1,73 +1,22 @@
...
-        return image, target
+import albumentations as A
+from albumentations.pytorch import ToTensorV2
 
 
 def get_transform(train=False):
-    transforms = []
-    transforms.append(Resize())
-    transforms.append(ToTensor())
+    augments = []
```
Comment (karolmajek) [inline semantic_segmentation/scripts/augment.py]: `augs` or `augmentations` are commonly used as names for a list of augmentations

File: semantic_segmentation/scripts/augment.py [hunk 2]
```diff
@@ -1,73 +1,22 @@
...
     if train:
-        transforms.append(RandomHorizontalFlip(0.5))
-        transforms.append(RandomRotate(degrees=30))
-    return Compose(transforms)
+        augments.append(A.RandomCrop(width=190, height=275, p=0.3))
+        augments.append(A.HorizontalFlip(p=0.5))
+        augments.append(A.RandomRotate90(p=0.5))
```
Comment (karolmajek) [inline semantic_segmentation/scripts/augment.py]: it will be better to use shiftscalerotate
and set rotate to <30 e.g. 20 (images are not perfectly in nadir, so large rotations will differ from test data)

Comment (karolmajek) [reply semantic_segmentation/scripts/augment.py]: shift -> 0.2 or 0.3 will be good
scale is a good idea (zoom out and zoom in) but hard to say the numbers -> check in nb or in albu demo (on their homepage)
you know that you can specify:
- single value e.g. `x` ->  it will be a range from -x to x
- or a tuple e.g (-0.3, 0.5) -> you will need this approach for the scale param

File: semantic_segmentation/scripts/augment.py [hunk 3]
```diff
@@ -1,73 +1,22 @@
...
+
+    augments.append(A.Resize(190, 275))
```
Comment (karolmajek) [inline semantic_segmentation/scripts/augment.py]: consider using `*size` instead of `190, 275`
common practices:
- use square input (let's ignore it for now; maybe later, but not smaller than 275)
- use size as param

File: semantic_segmentation/scripts/dataset.py [hunk 4]
```diff
@@ -29,13 +29,12 @@ def __getitem__(self, idx):
...
-        # img = img.resize((190, 275))
         target = Image.open(target_path).convert("L")
         target = np.array(target)
-        # target = target.resize((190, 275))
         if self.transforms is not None:
-            img, target = self.transforms(img, target)
-
+            transformed = self.transforms(image=img, mask=target)
+            img = transformed['image']
+            target = transformed['mask']
```
Comment (karolmajek) [inline semantic_segmentation/scripts/dataset.py]: good!

sometimes you want to just iterate ds without the transform - here you can do it without problem

File: semantic_segmentation/scripts/train.py [hunk 5]
```diff
@@ -24,14 +24,15 @@
 def show_results(imgs, preds, gts):
     fig, ax = plt.subplots(frameon=False)
     fig.set_size_inches(6, 36)
-    colormap = np.array([(16, 64,16), (255,0,0), (0,255,0), (0,0,255),
-                         (255,255,0), (255,0,255), (0,255,255), (255,255,255),
-                         (128,64,16), (128,16,64), (64,16,128), (16,64,128),
-                         (32,64,16), (32,16,64), (64,16,32), (16,64,32)
-                        ], dtype=np.uint8)
+    colormap = np.array([(16, 64, 16), (255, 0, 0), (0, 255, 0), (0, 0, 255),
```
Comment (karolmajek) [inline semantic_segmentation/scripts/train.py]: in such places I like to put constant number of elements per line e.g. 4 lines with 4 elements
it's easier to check/verify the code quickly

File: semantic_segmentation/scripts/train.py [hunk 6]
```diff
@@ -137,13 +138,13 @@ def train(model, train_ds, val_ds, optimizer, writer,
...
 if __name__ == '__main__':
-    unique_name=datetime.now().strftime("%Y_%m_%d-%H_%M_%S")
+    unique_name = datetime.now().strftime("%Y_%m_%d-%H_%M_%S")
     writer = SummaryWriter('runs/u2net/'+unique_name)
     parser = argparse.ArgumentParser()
     parser.add_argument('--datafolder', type=str, default="")
     args = parser.parse_args()
     model = U2NET_lite()
-    #optimizer = optim.Adam(model.parameters(), lr=0.005)
+    # optimizer = optim.Adam(model.parameters(), lr=0.005)
```
Comment (karolmajek) [inline semantic_segmentation/scripts/train.py:160]: parametrization -> later

**Review by karolmajek — CHANGES_REQUESTED** [review]
please change:
- augmentation related stuff - as commented in the code

additional idea:
- run `black` to enforce PEP8 and other formatiing rules for all files
- just seen that it can be done automatically pre-commit or post-commit on github

File: semantic_segmentation/scripts/train.py (hunk 5 above)
Comment (amrukwa) [reply semantic_segmentation/scripts/train.py]: black went for 1 per line, so i will leave it this way; 4 per line is too long, it turns out

File: semantic_segmentation/scripts/train.py (hunk 6 above)
Comment (amrukwa) [reply semantic_segmentation/scripts/train.py:160]: from command line or just generally?

Comment (karolmajek) [reply semantic_segmentation/scripts/train.py:160]: cmd or better - config file

let's try to use lightning params - they have some extra functionality for this

check it here:
https://pytorch-lightning.readthedocs.io/en/stable/common/hyperparameters.html

File: semantic_segmentation/scripts/augment.py [hunk 7]
```diff
@@ -10,6 +10,7 @@ def get_transform(train=False):
         augments.append(A.RandomRotate90(p=0.5))
 
     augments.append(A.Resize(190, 275))
+    augments.append(A.Normalize(mean=(0, 0, 0), std=(1, 1, 1)))
```
Comment (karolmajek) [inline semantic_segmentation/scripts/augment.py]: please don't!
there are only 2 popular values for those:
1. imagenet mean, std
2. (0.5, 0.5, 0.5) for both

File: semantic_segmentation/scripts/train.py [hunk 8]
```diff
@@ -147,7 +147,7 @@ def train(model, train_ds, val_ds, optimizer, writer,
     # optimizer = optim.Adam(model.parameters(), lr=0.005)
     optimizer = optim.RAdam(model.parameters(), lr=3e-4)
 
-    train_ds = Dataset(args.datafolder, get_transform())
+    train_ds = Dataset(args.datafolder, get_transform(True))
```
Comment (karolmajek) [inline semantic_segmentation/scripts/train.py]: put the name of the param explicitly - `train=True`

**Review by karolmajek — CHANGES_REQUESTED** [review]

File: semantic_segmentation/scripts/augment.py (hunk 2 above)
Comment (karolmajek) [reply semantic_segmentation/scripts/augment.py]: @amrukwa can you replace rotate90 with shiftScaleRotate?

File: semantic_segmentation/scripts/augment.py (hunk 7 above)
Comment (amrukwa) [reply semantic_segmentation/scripts/augment.py]: should i go for the second one then? cause i didn't want to scale the data in pytorch later on additionally and the new function for converting to tensors doesn't convert to floats

Comment (karolmajek) [reply semantic_segmentation/scripts/augment.py]: whatever - you don't need another normalization
expect your normalized data between -2..+2 after this normalization
if you chose 2nd you can _denormalize_ by multiplying by 0.5 and then adding 0.5 to the image (result will be approx 0..1)

File: semantic_segmentation/scripts/train.py (hunk 6 above)
Comment (amrukwa) [reply semantic_segmentation/scripts/train.py:160]: okay, so cmd for now, then integration in next step

File: semantic_segmentation/scripts/augment.py (hunk 2 above)
Comment (amrukwa) [reply semantic_segmentation/scripts/augment.py]: yes, on it, checking the values for scale for now

File: semantic_segmentation/scripts/augment.py [hunk 9]
```diff
@@ -1,73 +1,25 @@
...
+    return A.Compose(augs)
 
 
 def get_test_transform():
-    transforms = []
-    transforms.append(T.Resize((190, 275)))
-    transforms.append(T.ToTensor())
-    return T.Compose(transforms)
+    augs = []
+    # augs.append(A.Resize(190, 275))
```
Comment (karolmajek) [inline semantic_segmentation/scripts/augment.py]: in general test transform should be the same as the val one
resize is a good idea here

File: semantic_segmentation/scripts/dataset.py [hunk 10]
```diff
@@ -14,28 +14,28 @@ def __init__(self, root, transforms, train=True):
         self.imgs = list(sorted(os.listdir(os.path.join(root, "inputs"))))
         count = len(self.imgs)
         if train:
-            self.imgs = self.imgs[int(count*0.8):]
-            self.targets = list(sorted(os.listdir(os.path.join(root,
-                                "semantic_annotations"))))[int(count*0.8):]
+            self.imgs = self.imgs[int(count * 0.8) :]
+            self.targets = list(
+                sorted(os.listdir(os.path.join(root, "semantic_annotations")))
+            )[int(count * 0.8) :]
```
Comment (karolmajek) [inline semantic_segmentation/scripts/dataset.py:20]: warning - in most cases it is a bad idea to use 80:20 this way (better - `train_test_split` with frozen seed, or a list of files)

**Review by karolmajek — APPROVED** [review]
